src/python/pants/option/options.py

```diff
@@ -5,16 +5,55 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
 import copy
+import re
 import sys
 from builtins import object, open, str
 
-from pants.base.deprecated import warn_or_error
+from pants.base.deprecated import get_frame_info, warn_or_error
 from pants.option.arg_splitter import GLOBAL_SCOPE, ArgSplitter
 from pants.option.global_options import GlobalOptionsRegistrar
 from pants.option.option_util import is_list_option
 from pants.option.option_value_container import OptionValueContainer
 from pants.option.parser_hierarchy import ParserHierarchy, all_enclosing_scopes, enclosing_scope
 from pants.option.scope import ScopeInfo
+from pants.util.memo import memoized_property
+from pants.util.objects import datatype
+
+
+class CallableWrapper(datatype(['callable_object'])):
+  def __call__(self, *args, **kwargs):
+    return self.callable_object(*args, **kwargs)
+
+  def __new__(cls, callable_object):
+    if isinstance(callable_object, cls):
+      return callable_object
+    # `callable()` was deprecated in Python 3.0, then undeprecated in 3.2.
+    if not callable(callable_object):
+      cls.make_type_error("{!r} must be callable".format(callable_object))
+    return super(CallableWrapper, cls).__new__(cls, callable_object)
+
+
+class DeprecatedFlagMatcher(datatype([('scope_flags_fun', CallableWrapper)])):
```

Comment:
If you think this class is worth keeping (vs maybe inlining as static/class methods?), please add docstrings for it. It's good to think about will still be useful after the deprecation.

Reply:
Removed, see fb95e63972ee356bc3d1a7c23b361a88f2dae720!